api/v1alpha1/openstackrouter_types.go

```diff
@@ -0,0 +1,177 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// OpenStackRouterSpec defines the desired state of OpenStackRouter
+type OpenStackRouterSpec struct {
+	// Cloud is the OpenStackCloud hosting this resource
+	Cloud string `json:"cloud"`
+
+	// ID is the OpenStack UUID of the resource. If left empty, the
+	// controller will create a new resource and populate this field. If
+	// manually populated, the controller will adopt the corresponding
+	// resource.
+	ID string `json:"id,omitempty"`
+
+	// Name of the OpenStack resource.
+	Name string `json:"name,omitempty"`
+
+	Description string `json:"description,omitempty"`
+
+	ExternalGateway string `json:"externalGateway,omitempty"`
```

Comment:
This is called `external_gateway_info` in the API [docs](https://docs.openstack.org/api-ref/network/v2/#create-router). It should be `externalGatewayInfo` here for consistency

Reply:
That's strange. What the API is referring to is the ExternalGatewayInfo structure that contains a set of things. The externalGateway here is just a kubernetes name reference to a network, not an actual external_gateway_info object.

It tries to mimick the parameter that is passed on the CLI: _openstack router create vrouter --external-gateway public_

Reply:
what I'd do, is that I'd run the same openstackclient command with `--debug` to see how `public` is passed

Reply:
I have just tried it and it does the same that the controller does: it retrieves the network by name and wraps the NetworkID inside an external_gateway_info. 

The wrapper: ttps://github.com/gophercloud/gopherkube/pull/47/files#diff-1b8b81f0722366512bb81963c60c8f7f2fed15d6f3a42d8f3ad42489d62ff43eR168

Reply:
Perfect. Then I'd expose here the same struct that is exposed in the API, leaving the wrapping to the user